src/java/src/main/java/org/wololo/flatgeobuf/PackedRTree.java

```diff
@@ -336,10 +338,10 @@ public static SearchResult search(InputStream stream, int start, int numItems, i
         double maxY = rect.getMaxY();
         ArrayList<Integer> levelEnds = generateLevelEnds(numItems, nodeSize);
         int numNodes = levelEnds.get(0);
-        Stack<StackItem> stack = new Stack<StackItem>();
-        stack.add(new StackItem(0, levelEnds.size() - 1));
-        while (stack.size() != 0) {
-            StackItem stackItem = stack.pop();
+        PriorityQueue<StackItem> queue = new PriorityQueue<>((a, b) -> (int) (a.nodeIndex - b.nodeIndex));
+        queue.add(new StackItem(0, levelEnds.size() - 1));
```

Comment:
I don't disagree that a PriorityQueue should be faster than a stack in this case. But I'm curious now...

Since we're traversing the tree breadth first, that should be equivalent to just a regular queue - where we push onto the back and pop off the front. Right?

The [typescript implementation](https://github.com/flatgeobuf/flatgeobuf/blob/ec48353b169e41947a1923b9560494916ad5ed75/src/ts/packedrtree.ts#L144) actually use a straight-up queue like this.

Interestingly, the [CPP](https://github.com/flatgeobuf/flatgeobuf/blob/ec48353b169e41947a1923b9560494916ad5ed75/src/cpp/packedrtree.cpp#L341) and [rust](https://github.com/flatgeobuf/flatgeobuf/blob/ec48353b169e41947a1923b9560494916ad5ed75/src/rust/src/packed_r_tree.rs#L452) implementations use on ordered map - which actually seems wrong to me since [unordered_map doesn't guarantee any sorting](https://cplusplus.com/reference/unordered_map/unordered_map/begin/), right?

I might be missing something here though - I haven't looked into that closely.

Reply:
I did not look into it closely either, but I can tell you that at the beginning I had the direction of comparison flipped and tests were not passing anymore (did not check if the test was too rigid or it was a legit failure).

The core of the change is switching from sorting Stack, which uses TimSort with a computational complexity of O(n log n), towards inserting in a priority queue, which should have a complexity of O(log n).

Reply:
I wasn't aware Stack was sorted? Can't figure that from docs either (though it does say Deque should be used instead). The original implementation is from flatbush which most closely resembles what I implemented for TypeScript... that is, I don't think ordering is important here.

Reply:
Stack by itself is not sorted. It was getting sorted after every insert, that was the bit that slowed everything down a lot.